lib/Pandoc.pm

```diff
@@ -57,12 +59,19 @@ sub run {
     my $opts   = ref $_[-1] ? pop @_ : {};
     my @args   = @_;
 
-    my $in  = $opts->{in};
-    my $out = $opts->{out};
-    my $err = $opts->{err};
+    my %opts = ( %$opts, return_if_system_error => 1 );
+    my $in  = $opts{in};
+    my $out = $opts{out};
+    my $err = $opts{err};
+    for my $io ( qw[ in out err ] ) {
+        $opts{"binmode_std$io"} //= $opts{binmode} if $opts{binmode};
+        if ( 'SCALAR' eq ref $opts{$io} ) {
+            next unless utf8::is_utf8(${$opts{$io}});
+            $opts{"binmode_std$io"} //= ':utf8'; # or better :encoding(UTF-8) ?
```

Comment:
It's a UI thing really. On the one hand there can be any number of args to pass on to pandoc so either the options need to be a hashref or the pandoc args need to be an arrayref (or both...), but at the same time I don't want to modify the user's hashref so I copy it before adding the 'automatic' keys. It's either that or localizing the keys, and then I much prefer copying! :-) Surely calling the copy `%run3_opts` or somesuch might be less confusing. Note that if the `binmode_std*` keys are copied individually one may need to check for their existence in the user-supplied options, depending on how run3() treats existing keys with undef values.

Thinking it over the better UI is probably to pass the pandoc args in an arrayref: they are likely to remain the same across multiple calls while the in/out/err scalars will be different.

As for `:encoding(UTF-8)` vs. `:utf8` the former is safer and the latter is way faster.